detoks/src/main/java/nl/tudelft/trustchain/detoks/gossiper/ProfileEntryGossiper.kt

```diff
@@ -32,28 +33,28 @@ class WatchTimeGossiper(
 
         val randomPeers = pickRandomN(deToksCommunity.getPeers(), peers)
         val randomProfileEntries = pickRandomN(
-            TorrentManager.getInstance(context).profile.profiles.entries.map { Pair(it.key, it.value.watchTime) },
+            TorrentManager.getInstance(context).profile.profiles.entries.map { Pair(it.key, it.value) },
             blocks
         )
         if (randomPeers.isEmpty() || randomProfileEntries.isEmpty()) return
 
         randomPeers.forEach {
             deToksCommunity.gossipWith(
                 it,
-                WatchTimeMessage(randomProfileEntries),
-                DeToksCommunity.MESSAGE_WATCH_TIME_ID
+                ProfileEntryMessage(randomProfileEntries),
+                DeToksCommunity.MESSAGE_PROFILE_ENTRY_ID
             )
         }
     }
 }
 
-class WatchTimeMessage(data: List<Pair<String, Long>>) : GossipMessage<Long>(data) {
-    companion object Deserializer : Deserializable<WatchTimeMessage> {
-        override fun deserialize(buffer: ByteArray, offset: Int): Pair<WatchTimeMessage, Int> {
+class ProfileEntryMessage(data: List<Pair<String, ProfileEntry>>) : GossipMessage<ProfileEntry>(data) {
+    companion object Deserializer : Deserializable<ProfileEntryMessage> {
+        override fun deserialize(buffer: ByteArray, offset: Int): Pair<ProfileEntryMessage, Int> {
             val msg = deserializeMessage(buffer, offset){
-                return@deserializeMessage Pair(it.getString(0), it.getLong(1))
+                return@deserializeMessage Pair(it.getString(0), it.get(1) as ProfileEntry)
```

Comment:
Did you test this with 2 peers? Because I would expect that deserialize would crash with `it.get(1) as ProfileEntry`

Reply:
Fixed this as suggested during the meeting but only for entries, not for sharing whole profiles as this would require us to work with jsonobjects/arrays instead of a string and we are too constrained on time. Instead I just send more messages.